tests/app/main/test_services.py

```diff
@@ -715,18 +934,33 @@ def test_update_with_under_50_words_error(self, data_api_client):
 
     def test_update_non_existent_service_returns_404(self, data_api_client):
         data_api_client.get_service.return_value = None
-        res = self.client.post('/suppliers/services/1/edit/description')
+        res = self.client.post('/suppliers/frameworks/g-cloud-6/services/1/edit/description')
 
         assert res.status_code == 404
         self.assert_no_flashes()
+        assert data_api_client.update_service.called is False
 
     def test_update_non_existent_section_returns_404(self, data_api_client):
         data_api_client.get_service.return_value = self.empty_service
         res = self.client.post(
-            '/suppliers/services/1/edit/invalid_section'
+            '/suppliers/frameworks/g-cloud-6/services/1/edit/invalid_section'
         )
         assert res.status_code == 404
         self.assert_no_flashes()
+        assert data_api_client.update_service.called is False
+
+    def test_incorrect_framework_fails(self, data_api_client):
```

Comment:
One's a `GET`, the other's a `POST`, and yes, calling the fixture `empty_service` is a bit ambiguous (wasn't me) - what it actually means is a perfectly valid service that doesn't really have any content.